cmd/celvet/main.go

```diff
@@ -80,24 +90,45 @@ func main() {
 		os.Exit(1)
 	}
 
-	limitErrors := celvet.CheckMaxLimits(structural)
-	for _, lintError := range limitErrors {
-		fmt.Fprintf(os.Stderr, "%s\n", lintError)
+	lintErrs := celvet.NewLinterErrors()
+	lintErrs.AddLinterErrors(celvet.CheckMaxLimits(structural))
+	lintErrs.AddLinterErrors(celvet.CheckExprCost(structural))
+	if useJSON {
+		err := emitJSON(lintErrs)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error generating JSON output: %s\n", err)
+			os.Exit(1)
+		}
+	} else {
+		emitText(lintErrs, *humanReadable)
 	}
 
-	costErrors, compileErrors := celvet.CheckExprCost(structural)
-	for _, lintError := range costErrors {
-		if *humanReadable {
+	if lintErrs.HasErrors() {
+		os.Exit(1)
+	}
+}
+
+func emitJSON(l *celvet.LinterErrors) error {
+	encoder := json.NewEncoder(os.Stdout)
```

Comment:
Why stdout for JSON format and stderr for text format?

Reply:
`stderr` for `emitText` was a mistake on my part, sorry (or at least if it wasn't, I can't recall the rationale/didn't record it) - fixed